app/src/main/java/org/schabi/newpipe/util/external_communication/InternalUrlsHandler.java

```diff
@@ -49,22 +49,20 @@ public static boolean handleUrl(final Context context,
                                     final int timestampType) {
         String matchedUrl = "";
         int seconds = -1;
-        final Pattern timestampPattern;
+        final Matcher matcher;
 
         if (timestampType == 0) {
-            timestampPattern = AMPERSAND_TIMESTAMP_PATTERN;
+            matcher = AMPERSAND_TIMESTAMP_PATTERN.matcher(url);
         } else if (timestampType == 1) {
-            timestampPattern = HASHTAG_TIMESTAMP_PATTERN;
+            matcher = HASHTAG_TIMESTAMP_PATTERN.matcher(url);
         } else {
             return false;
         }
 
-        final Matcher matcher = timestampPattern.matcher(url);
```

Comment:
Having a common matcher actually made sense, I think

Reply:
This is an outdated code, fixed.